BuildPropertiesPlugin/plugin/src/main/groovy/com/novoda/buildproperties/Entry.groovy

```diff
@@ -34,4 +34,29 @@ class Entry {
         value.call()
     }
 
+    Entry or(def fallback) {
+        def other = from(fallback)
+        new Entry(key, {
+            try {
+                return getValue()
+            } catch (Throwable e) {
+                try {
+                    return other.call()
+                } catch (Throwable e2) {
+                    throw CompositeException.from(e).add(e2)
```

Comment:
Yes, that would be too much effort. Isn't the cause implicit if you do `new Exception(cause)`?